src/components/diagrams/diagram-pane.js

```diff
@@ -286,95 +362,314 @@ export function DiagramPane({
+                );
+                if (!sameNadAlreadyPresentInViews) {
+                    addOrReplaceNAD(
+                        networkAreaIds,
+                        networkAreaViewState,
+                        networkAreaDiagramDepth
+                    );
+                }
+            } else if (
+                viewsRef.current.find(
+                    (diagramView) =>
+                        diagramView.svgType === DiagramType.NETWORK_AREA_DIAGRAM
+                )
+            ) {
+                // no more NAD, remove it from the views
+                removeNAD();
             }
+        },
+        [addOrReplaceNAD, removeNAD, networkAreaDiagramDepth]
+    );
+
+    // Update only the state of the diagrams (opened, minimized, etc)
+    const updateDiagramStates = useCallback((diagramStates) => {
+        // We check if we need to update some diagrams
+        let diagramsToUpdate = [];
+        diagramStates.forEach((diagramState) => {
+            // if SLD
+            if (diagramState.svgType !== DiagramType.NETWORK_AREA_DIAGRAM) {
+                const diagramIndex = viewsRef.current.findIndex(
+                    (diagramView) =>
+                        diagramView.svgType !==
+                            DiagramType.NETWORK_AREA_DIAGRAM &&
+                        diagramView.id === diagramState.id &&
+                        diagramView.state !== diagramState.state
+                );
+                // if we found a diagram with a new state (ex : minimized)
+                if (diagramIndex > -1) {
+                    diagramsToUpdate.push({
+                        index: diagramIndex,
+                        state: diagramState.state,
+                    });
+                }
+            }
+            // if NAD
+            else {
+                // no need to check the ID because we have only one NAD in the views
+                // diagramIndex can only be -1 (if no match) or viewsRef.current.length - 1 (if match)
+                const diagramIndex = viewsRef.current.findIndex(
+                    (diagramView) =>
+                        diagramView.svgType ===
+                            DiagramType.NETWORK_AREA_DIAGRAM &&
+                        diagramView.state !== diagramState.state
+                );
+                // if the NAD has a new state (all NAD are supposed to have the same state)
+                if (
+                    diagramIndex > -1 &&
+                    // we don't want to add it twice
+                    !diagramsToUpdate.find(
+                        (diagram) => diagram.index === diagramIndex
+                    )
+                ) {
+                    diagramsToUpdate.push({
+                        index: diagramIndex,
+                        state: diagramState.state,
+                    });
+                }
+            }
+        });
+
+        // If we found diagrams to update, we update them
```

Comment:
// If we found diagrams to update, we update their states in "views"

Reply:
It will be duplicate with the comment above